src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseBase.cpp

```diff
@@ -58,6 +58,17 @@ SinglePhaseBase::SinglePhaseBase( const string & name,
     setApplyDefaultValue( 0.0 ).
     setInputFlag( InputFlags::OPTIONAL ).
     setDescription( "Temperature" );
+
+  this->registerWrapper( viewKeyStruct::maxPressureChangeString(), &m_maxPressureChange ).
+    setSizedFromParent( 0 ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setApplyDefaultValue( -1.0 ).   // disabled by default
+    setDescription( "Maximum (absolute) pressure change in a Newton iteration" );
+
+  this->registerWrapper( viewKeyStruct::allowNegativePressureString(), &m_allowNegativePressure ).
+    setApplyDefaultValue( 0 ). // by default not allowed
```

Comment:
doesn't this happen quite often in hydrafrac simulations? Maybe we should have the hydrofrac solver change this value to 1 by default.

Reply:
yes it happens routinely there and hydrofrac solver does enable that flag https://github.com/GEOS-DEV/GEOS/blob/7025b62207f5dad87edd8bb8d0f03df2be4ef46e/src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp#L175, let me clarify the comment